### PR TITLE
[1] Make trf compare in mudensity gui not need Mosaiq access

### PR DESCRIPTION
**Pull request by SimonBiggs**
@sjswerdloff (as always no urgency to review during designated down time :) )

This is to make it so that TRF comparison can be done without Mosaiq access.

### CODE REVIEW

File: pymedphys/_gui/streamlit/mudensity-compare.py [hunk 1]
```diff
@@ -814,33 +818,46 @@ def trf_input_method(patient_id="", key_namespace="", **_):
 
     headers
 
+    deliveries = cached_deliveries_loading(tables, delivery_from_trf)
+
+    individual_identifiers = [
```
Comment (sjswerdloff) [inline pymedphys/_gui/streamlit/mudensity-compare.py:823]: patient id ?  DICOM Patient ID, MOSAIQ PAT_ID1 ?
a comment here would be helpful

File: pymedphys/_gui/streamlit/mudensity-compare.py [hunk 2]
```diff
@@ -814,33 +818,46 @@ def trf_input_method(patient_id="", key_namespace="", **_):
...
+    except KeyError:
+        use_mosaiq = False
+        st.write(
+            NoMosaiqAccess(
+                "Need Mosaiq access to determine patient name. "
+                "Patient name set to 'Unknown'."
+            )
         )
-        patient_names.add(patient_name)
+        patient_name = "Unknown"
```
Comment (sjswerdloff) [inline pymedphys/_gui/streamlit/mudensity-compare.py:845]: I would set this above the GUI part, and then pass in patient_name to the GUI.
Could also put the identifier that is known in to the GUI so it's clear what is happening to which data.

File: pymedphys/_gui/streamlit/mudensity-compare.py [hunk 3]
```diff
@@ -814,33 +818,46 @@ def trf_input_method(patient_id="", key_namespace="", **_):
...
 
-    patient_name = filter_patient_names(patient_names)
+    if use_mosaiq:
+        mosaiq_details = mosaiq_details.drop("beam_completed", axis=1)
```
Comment (sjswerdloff) [inline pymedphys/_gui/streamlit/mudensity-compare.py:848]: suggest extract method

File: pymedphys/_gui/streamlit/mudensity-compare.py [hunk 4]
```diff
@@ -1271,7 +1288,7 @@ def run_calculation(
         )
 
     st.write("## Results")
-    st.pyplot()
+    st.pyplot(fig)
```
Comment (sjswerdloff) [inline pymedphys/_gui/streamlit/mudensity-compare.py:1291]: note that the API from imagemagick has changed so any call to magick is at risk of failing if someone installs the newer version.  Suggest creating a command lookup that toggles based on which API is available.

File: setup.py [hunk 5]
```diff
@@ -219,7 +219,7 @@
 
 setup_kwargs = {
     "name": "pymedphys",
-    "version": "0.34.0.dev2",
+    "version": "0.34.0.dev3",
```
Comment (sjswerdloff) [inline setup.py:222]: inconsistent with semver and above representations, which are 0.34.0-dev3

**Review by sjswerdloff — CHANGES_REQUESTED** [review]
setup.py is inconsistent with semver.
If that is necessary, play through.
Urge review of the other comments though.

File: setup.py (hunk 5 above)
Comment (SimonBiggs) [reply setup.py:222]: > setup.py is inconsistent with semver.
> If that is necessary, play through.
> Urge review of the other comments though.

Thanks for the review :). I really, really appreciate these. Thank you. Making my dream of peer review in medical physics open source software a reality :). Thank you.

**Comment by SimonBiggs** [comment]
> setup.py is inconsistent with semver.
> If that is necessary, play through.
> Urge review of the other comments though.

Thanks for the review :). I really, really appreciate these. Thank you. Making my dream of peer review in medical physics open source software a reality :). Thank you.

File: setup.py (hunk 5 above)
Comment (SimonBiggs) [reply setup.py:222]: For this one, this is actually the Python default... And it is what poetry autogenerates. So I'll leave this one here.

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 4 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py:1291]: Yup, good call. Will fix.

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 3 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py:848]: Yes, good call. Also, this whole file is too huge. ... it's unwieldly. It got that way due to streamlit not autoreloading on imports, so to keep the autoreloading I kept it in the same file. ...it's painful though.

I won't correct the file issue here, but I'll begin that by pulling this block out.

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 2 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py:845]: A little unsure what you mean here. I'll think on it more when I'm at a computer.

**Comment by SimonBiggs** [comment]
Also, gave a talk yesterday, here's the recording:

https://youtu.be/_HeJBfkXfxg

Kinda cool being able to present this stuff. Thank you for making the with me @sjswerdloff :)

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 1 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py:823]: Yup, these TRF files are indexed by MOSAIQ `Ident.PAT_ID1` during collection in our centre. That is what this Patient ID refers to.

See https://github.com/pymedphys/pymedphys/blob/f4a5efd41d0f48907686d3ee562ec1c3bf42fadb/pymedphys/_mosaiq/delivery.py#L122-L123

Will add comment.

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 2 above)
Comment (sjswerdloff) [reply pymedphys/_gui/streamlit/mudensity-compare.py:845]:
```
        use_mosaiq = False
        patient_name = "Unknown"
        st.write(
            NoMosaiqAccess(
                "Need Mosaiq access to determine patient name. "
                f"Patient name set to '{patient_name}'."
            )
        )
```

Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py:861]: In practice, the user also provides a reference set of data (Monaco, Mosaiq, or RT plan DICOM). The GUI happily extracts the patient name from the reference and having the evaluation dataset be missing the name at the end of the day has no impact.

Can we revisit this one after this PR?

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 3 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py]: So, I wanted to pull out `trf_input_method` into its own file, and then pull out this `use_mosaiq` block into its own function. I did that, but the flow on refactoring effect was large due to the need to pull out shared functions.

To make it easier to find, here is the function refactored out as suggested:

https://github.com/pymedphys/pymedphys/blob/853cca22de62e837286812408b9af7997d63f6f9/pymedphys/_gui/streamlit/mudensity/_trf.py#L34-L67

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 4 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py]: Yup, this is done now. See the implementation here:

https://github.com/pymedphys/pymedphys/blob/651d7aa7d7c4b25f884ce01bb32f80424a9a03eb/pymedphys/_gui/streamlit/mudensity/main.py#L956-L1005

**Comment by SimonBiggs** [comment]
@sjswerdloff Well that exploded :). I believe I have now addressed each of your comments, so I shall use the "re-request review" button.

File: pymedphys/_gui/streamlit/mudensity-compare.py (hunk 1 above)
Comment (SimonBiggs) [reply pymedphys/_gui/streamlit/mudensity-compare.py]: Just so it's easier to find here is a link to the added docstring:

https://github.com/pymedphys/pymedphys/blob/651d7aa7d7c4b25f884ce01bb32f80424a9a03eb/pymedphys/_gui/streamlit/mudensity/_trf.py#L70-L86

**Comment by SimonBiggs** [comment]
I should have done the refactor in a separate PR so that this was easier to review :(

**Comment by SimonBiggs** [comment]
> I should have done the refactor in a separate PR so that this was easier to review :(

Here, let me do that. I'll make a separate PR where nothing is changed, just functions moved around. Then this one will make it clear what's changed in the review panel.

**Comment by SimonBiggs** [comment]
Hi @sjswerdloff,

Sorry about the previous state of the "files changed" tab. I have now done a "move functions only" PR over at https://github.com/pymedphys/pymedphys/pull/1118. And when I use that as the base for this PR all of a sudden the "files changed" tab is far easier to read and is actually able to sensibly reviewed.

**Comment by SimonBiggs** [comment]
Just a heads up @sjswerdloff, so that it's not off the radar. I believe the changes requested here have been addressed.

**Review by sjswerdloff — APPROVED** [review]

**Pull request merged**